src/main/java/com/openlattice/chronicle/services/upload/AppDataUploadManagerImpl.java

```diff
@@ -0,0 +1,654 @@
+                ApiUtil.generateDefaultEntityId( ImmutableList.of( commonTasksManager.getPropertyTypeId( OL_ID_FQN ) ),
+                        entityData )
+        );
+    }
+
+    private EntityKey getHasEntityKey( UUID hasESID, Map<UUID, Set<Object>> entityData ) {
+
+        return new EntityKey(
+                hasESID,
+                ApiUtil.generateDefaultEntityId( ImmutableList.of( commonTasksManager.getPropertyTypeId( OL_ID_FQN ) ),
+                        entityData )
+        );
+    }
+
+    // HELPER METHODS: logData
+    private Map<UUID, Set<Object>> getUsedByEntity( String appPackageName, String dateLogged, String participantId ) {
+        Map<UUID, Set<Object>> entity = new HashMap<>();
+
+        entity.put( commonTasksManager.getPropertyTypeId( DATE_TIME_FQN ), ImmutableSet.of( dateLogged ) );
+        entity.put( commonTasksManager.getPropertyTypeId( FULL_NAME_FQN ), ImmutableSet.of( appPackageName ) );
+        entity.put( commonTasksManager.getPropertyTypeId( PERSON_ID_FQN ), ImmutableSet.of( participantId ) );
+
+        return entity;
+    }
+
+    private Map<UUID, Set<Object>> getRecordedByEntity( String deviceId, String appPackageName, String dateLogged ) {
+        Map<UUID, Set<Object>> entity = new HashMap<>();
+
+        entity.put( commonTasksManager.getPropertyTypeId( FULL_NAME_FQN ), ImmutableSet.of( appPackageName ) );
+        entity.put( commonTasksManager.getPropertyTypeId( DATE_LOGGED_FQN ), ImmutableSet.of( dateLogged ) );
+        entity.put( commonTasksManager.getPropertyTypeId( STRING_ID_FQN ), ImmutableSet.of( deviceId ) );
+
+        return entity;
+    }
+
+    private Map<UUID, Set<Object>> getHasEntity( String firstDateTime ) {
+        Map<UUID, Set<Object>> entity = Maps.newHashMap();
+        entity.put( commonTasksManager.getPropertyTypeId( OL_ID_FQN ), Set.of( firstDateTime ) );
+
+        return entity;
+    }
+
+    private Map<UUID, Set<Object>> getMetadataEntity( UUID participantEKID ) {
+        Map<UUID, Set<Object>> entity = Maps.newHashMap();
+        entity.put( commonTasksManager.getPropertyTypeId( OL_ID_FQN ), Set.of( participantEKID ) );
+
+        return entity;
+    }
+
+    private Map<UUID, Set<Object>> getUserAppsEntity( String appPackageName, String appName ) {
+
+        Map<UUID, Set<Object>> entity = new HashMap<>();
+
+        entity.put( commonTasksManager.getPropertyTypeId( FULL_NAME_FQN ), ImmutableSet.of( appPackageName ) );
+        entity.put( commonTasksManager.getPropertyTypeId( TITLE_FQN ), ImmutableSet.of( appName ) );
+
+        return entity;
+    }
+
+    private EntityKey getDeviceEntityKey( UUID devicesESID, String deviceId ) {
+        return new EntityKey(
+                devicesESID,
+                deviceId
+        );
+    }
+
+    private EntityKey getParticipantEntityKey( UUID participantESID, String participantId ) {
+        return new EntityKey(
+                participantESID,
+                participantId
+        );
+    }
```

Comment:
I feel like these don't need to exist.

Reply:
why is that?

Reply:
They don't provide any extra value. You can just do `new EntityKey()` inline.

Reply:
Yes, BUT then we would have to make sure we are consistent across all places where the entity keys are used. It seems it would be really easy to lose track of all the corresponding updates in future refactors, thus accidentally breaking things.